### PR TITLE
Comment on PRs if a remote call to merge a change failed

### LINKED ISSUE #115: Zuul should report back to github on failure
So I was doing some testing with zuul on a repo that had branch protection enabled. The branch protection criteria was not met, but because we were using labels at the time i could trigger a bonny approved. 

I'm pretty sure zuul tried to perform the merge and everything but github rejected the push because the branch protection failed. 

It'll be way to difficult to make zuul read and understand github's branch protection (at least for now) but at the very least when a merge or push failure happens zuul should drop a comment to say that something went wrong, what happened and what to do next.

### PR DESCRIPTION
**Pull request by gandelman-a**
This adds a comment to PRs when a successfully tested change fails
to merge into an upstream repository due to a Github API error. It
uses the merge_failure_message to format said message.

closes BonnyCI/projman#115

Signed-off-by: Adam Gandelman <adamg@ubuntu.com>

### CODE REVIEW

**Comment by gandelman-a** [comment]
So I'm not sure we want to re-use the merge-failure-message pipeline config option or introduce something new to cover this specific type of failure.  The intended use-case for the existing config option is to format the messages that are posted back when a change fails to merge on the mergers (ie due to a merge conflict of the change under test).   Here we're posting back a message after all testing has passed but the remote github call to merge the thing upstream fails (ie because a required branch protection status has not been set).  We could add a new option 'remote-merge-failure-message' or something similar? Feedback welcome

**Comment by omgjlk** [comment]
I think this is a fine approach for 2.5, and for v3 we'll go the route of a specific failure case reporting for this, which would be different from mergers merging to begin with.

Waiting until after demo to approve.

**Review by omgjlk — APPROVED** [review]

**Pull request merged**
